tests/requirements_file_exercise_tests.py

```diff
@@ -0,0 +1,132 @@
+                                                 "typing-extensions==3.10.0.2"]
+        mock_lines_of_third_requirement_file = ["six==1.16.0", "Flask-RESTful==0.3.9", "redis==2.10.6"]
+        mock_read_file_lines.side_effect = [mock_lines_of_first_requirement_file, mock_lines_of_second_requirement_file,
+                                            mock_lines_of_third_requirement_file]
+        excepted = "Flask Jinja2 validators requests typing-extensions six Flask-RESTful redis"
+
+        actual = create_libraries_string(first_requirements_file)
+
+        mock_read_file_lines.assert_has_calls([call(first_requirements_file), call(second_requirements_file),
+                                               call(third_requirements_file)])
+        self.assertEqual(mock_read_file_lines.call_count, 3)
+        self.assertEqual(excepted, actual)
+
+    @mock.patch('src.requirements_file_exercise.read_file_lines')
+    def test_create_libraries_string_should_raise_file_not_found_error(self, mock_read_file_lines):
+        mock_read_file_lines.side_effect = FileNotFoundError
+        self.assertRaises(FileNotFoundError, create_libraries_string, "no_exist.txt")
+
+    @mock.patch('src.requirements_file_exercise.read_file_lines')
+    def test_create_libraries_string_should_add_libraries_names_without_redundant_spaces(self, mock_read_file_lines):
+        mock_read_file_lines.return_value = ["Flask==2.0.2   ", "Flask-RESTful  ==0.3.9", "Jinja2==  3.0.2",
+                                             "   validators==0.18.2"]
+        file_name = 'path_to_rquirements_file.txt'
+        excepted = "Flask Flask-RESTful Jinja2 validators"
+
+        actual = create_libraries_string(file_name)
+
+        self.assertEqual(excepted, actual)
+
+    @mock.patch('src.requirements_file_exercise.read_file_lines')
+    def test_should_create_libraries_string_from_infinite_recursion_files(self, mock_read_file_lines):
+        first_requirements_file = 'requirements.txt'
+        second_requirements_file = 'second-requirements.txt'
+        mock_lines_of_first_requirement_file = ["Flask==2.0.2", "Flask-RESTful==0.3.9", "Jinja2==3.0.2",
+                                                "-r " + second_requirements_file, "validators==0.18.2"]
+        mock_lines_of_second_requirement_file = ["requests==2.26.0", "six==1.16.0", "-r " + first_requirements_file,
+                                                 "typing-extensions==3.10.0.2"]
+        mock_read_file_lines.side_effect = [mock_lines_of_first_requirement_file, mock_lines_of_second_requirement_file]
+        excepted = "Flask Flask-RESTful Jinja2 validators requests six typing-extensions"
+
+        actual = create_libraries_string(first_requirements_file)
+
+        mock_read_file_lines.assert_has_calls([call(first_requirements_file), call(second_requirements_file)])
+        self.assertEqual(mock_read_file_lines.call_count, 2)
+        self.assertEqual(excepted, actual)
```

Comment:
You missed a few relevant use cases:
Add test the check the case for -replica==3.715 to check whats happened if do you have "-r" and no "-r " 
Add a test that you get libraries with no versions

Again think about the right way to write the tests... You don't need to write 100 tests for 100 use cases, you could write an amazing test with 20 tests (Or even less) that check 100 use-cases. 
etc...